packages/builder/src/components/automation/SetupPanel/CronBuilder.svelte

```diff
@@ -2,15 +2,36 @@
   import { Button, Select, Input, Label } from "@budibase/bbui"
   import { onMount, createEventDispatcher } from "svelte"
   import { flags } from "stores/backend"
+  import { helpers } from "@budibase/shared-core"
 
   const dispatch = createEventDispatcher()
 
   export let value
+  let error
+
+  const reboot = "@reboot"
+
+  $: {
+    const exists = CRON_EXPRESSIONS.some(cron => cron.value === value)
+    const customIndex = CRON_EXPRESSIONS.findIndex(
+      cron => cron.label === "Custom"
+    )
+
+    if (!exists && customIndex === -1) {
+      CRON_EXPRESSIONS[0] = { label: "Custom", value: value }
+    } else if (exists && customIndex !== -1) {
+      CRON_EXPRESSIONS.splice(customIndex, 1)
+    }
+  }
 
   const onChange = e => {
-    if (e.detail === value) {
+    if (value !== reboot) {
+      error = helpers.cron.validate(e.detail).err
+    }
+    if (e.detail === value || error) {
```

Comment:
NAB: Just wondering what this line does? Is it not possible to have both **value** and **error** set?

Reply:
If the cron value has not changed (e.detail === value) or there is an error, then we simply return early from the function

Reply:
Ah yes, thank you!

I was reading it like: `e.detail === (value || error)` for some reason.